app/javascript/packages/webauthn/enroll-webauthn-device.ts

```diff
@@ -37,6 +37,33 @@ interface EnrollResult {
   transports?: string[];
 }
 
+/**
+ * @see https://www.iana.org/assignments/cose/cose.xhtml#algorithms
+ */
+const enum COSEAlgorithm {
+  ES256 = -7,
+  ES384 = -35,
+  ES512 = -36,
+  PS256 = -37,
+  PS384 = -38,
+  PS512 = -39,
+  RS256 = -257,
+}
+
+/**
+ * @see https://github.com/18F/identity-idp/blob/main/config/initializers/webauthn.rb
+ * @see https://github.com/cedarcode/webauthn-ruby/blob/6db9596/lib/webauthn/relying_party.rb#L16
+ */
+const SUPPORTED_ALGORITHMS: COSEAlgorithm[] = [
+  COSEAlgorithm.ES256,
+  COSEAlgorithm.ES384,
+  COSEAlgorithm.ES512,
+  COSEAlgorithm.PS256,
+  COSEAlgorithm.PS384,
+  COSEAlgorithm.PS512,
+  COSEAlgorithm.RS256,
+];
```

Comment:
does the `const` optimization prevent us from doing this?
```suggestion
const SUPPORTED_ALGORITHMS: COSEAlgorithm[] = Object.values(COSEAlgorithm);
```

Reply:
ok turns out yes, `const` does prevent us from doing that :[ :[

Reply:
Yeah, to this and your other comment thread: I'd gone through a few iterations here to try to avoid duplicating the set and iterating instead. But as you noted, the inlining optimization prevents the iteration, and to not use `const enum` causes a lot of bloat in the generated output.

[See TypeScript Playground example](https://www.typescriptlang.org/play?#code/MYewdgzgLgBApmArgWxgYQPIGUCiBBAGwHMQAnASygAtk1xoclUBvAKBhhywCYBWANhgBeGAFoA7ABp2nLAGYAHABZhYub2kcuvAIzdVouf00wASjwEG5UmecUqRhhSfO79juQE4TABQuDHPhsAX1ZWUEhYLABVHx8MUwAVHAARAH08ABkAcQSASUSACQBZLDTMADksRLScCujigC50bHxiMkoaOkjGFABtAF1VPplMXEISCmpaeihe5AA6Lj5jUdaJjunuhiYl+WUTMbbJzpme3e09Q-X2qa7Z+YXzFevx29Ptud27A7W3k62D2+WDcr2Om3u5xQCz8L1YAwA3GFWAB6FGcUikMgAQlR6IwACMAFZwYBQBYANwAhgREHAIAAKI4bO5nHYoACUSNYCBQLX+EOQ8xgbC0-gMNjF9isGhkl3calWHGelg8krM+wcamcthBeis3hksNVYiC0lC4VmMBicQSyXSWVypgKJTKdQazWZ72m80Gwz+4NZj2WAjBLNOwc1Ye9NEjoID4Z9wLhHC9ANjwPs0fTQuB8dTNxzj2Nq0RYUJJLJlJpdMZacF8y5QA)

The way I ended up here, I was semi-intentional in separating `COSEAlgorithm` as an enum representing _all of the valid algorithms_, vs. `SUPPORTED_ALGORITHMS` as the subset of `COSEAlgorithm` we happen to support. For practicality's sake, I didn't actually include all of the possible values in `COSEAlgorithm`, especially since we don't care for the ones we don't support. But we _could_ include more/all values there.

Reply:
oh yikes a reverse mapping in there, I had no idea

thanks for walking through it!

Reply:
I pushed some revisions to the DocBlock comments in 94154e97c to clarify the intended distinction mentioned in my previous comment.